backstory/src/main/java/com/google/sps/story/data/NLServiceClient.java

```diff
@@ -0,0 +1,185 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.story.data;
+
+import com.google.cloud.language.v1.AnalyzeSyntaxResponse;
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.PartOfSpeech;
+import com.google.cloud.language.v1.PartOfSpeech.Form;
+import com.google.cloud.language.v1.PartOfSpeech.Proper;
+import com.google.cloud.language.v1.PartOfSpeech.Tag;
+import com.google.cloud.language.v1.Token;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Service client for Cloud Natural Language API
+ */
+public class NLServiceClient {
+  /** holds the language service client instance for this service client */
+  private LanguageServiceClient client;
+
+  /**
+   * Constructs a default object of client service class.
+   *
+   * @throws IOException if an instance of LanguageServiceClient could not be created
+   */
+  public NLServiceClient() throws IOException {
+    this(LanguageServiceClient.create());
+  }
+
+  /**
+   * Constructs a NLServiceClient class with a given LanguageServiceClient
+   * (mainly to be used for testing)
+   *
+   * @param client a given LanguageServiceClient to use to access Natural Language API
+   * @throws IllegalArgumentException if client is null
+   */
+  public NLServiceClient(LanguageServiceClient client) throws IllegalArgumentException {
+    if (client == null) {
+      throw new IllegalArgumentException("Client cannot be null");
+    }
+
+    this.client = client;
+  }
+
+  /**
+   * Takes a list of words and sorts them by word type.
+   * Returns a map which has word types as keys and then words
+   * in a list linked to their relevant word type (e.g. "Mary" would
+   * be sorted into the PROPER_NOUN list and "calm" would be sorted
+   * into the adjective list).
+   *
+   * @param words the words to sort by word type
+   * @return a map with lists of words sorted by their word type and with their
+   *    word type as a key
+   */
+  public Map<WordType, List<String>> sortByWordType(List<String> words) {
+    Map<WordType, List<String>> map = new HashMap<WordType, List<String>>();
+
+    List<String> singleWords = new ArrayList<String>();
+
+    // remove multi-word inputs first and store as multiword nouns
+    for (String text : words) {
+      if (text.contains(" ")) {
```

Comment:
Do other types of whitespace need to be considered?

Reply:
I don't think so. From working with the inputs that come up and how NL API filters, I believe that the onlly thing we want to remove is actual multiple word inputs. If they're hyphenated, I believe it would be considered one word.

Reply:
@ykjchen Is that sufficient?

Reply:
Hey, I was thinking more in terms of there being various whitespace characters:
https://stackoverflow.com/a/4067825